### PR TITLE
Ports: Add m4 port

### PR DESCRIPTION
**Pull request by MWGuy**

### CODE REVIEW

File: Ports/m4/package.sh [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash ../.port_include.sh
+port=m4
+version=1.4.9
+useconfigure=true
+files="http://ftp.gnu.org/gnu/m4/m4-1.4.9.tar.gz m4-1.4.9.tar.gz"
+
+configure() {
+    run ./configure --host=386bsd
```
Comment (bugaevc) [inline Ports/m4/package.sh]: mmm?

Comment (awesomekling) [inline Ports/m4/package.sh]: Why are we using `--host=386bsd` here?

Comment (MWGuy) [reply Ports/m4/package.sh]: Just override ``--host=i686-pc-serenity`` from ``.port_include.sh``, because ``i686-pc-serenity`` host needs a separate patch

**Comment by MWGuy** [comment]
Opss, with ``configopts`` works too

File: Ports/m4/package.sh (hunk 1 above)
Comment (awesomekling) [reply Ports/m4/package.sh]: But.. we are not a 386bsd system, so what is the significance of 386bsd here?

Comment (Redfoxymoon) [reply Ports/m4/package.sh]: Yeah, config.sub needs to be patched for the correct triplet to work, we should probably keep a fixed config.sub around and simply replace it in any ports that uses autotools, configuring for 386bsd is wrong regardless.

**Comment by MWGuy** [comment]
@awesomekling fix in https://github.com/SerenityOS/serenity/pull/856/commits/af85275c6ec3f55fd44f980a9833f55786c54c09

**Comment by awesomekling** [comment]
Sure :)

**Comment by awesomekling** [comment]
The fix for `wint_t` looks weird, but we can fix that down the road.

**Pull request merged**